test/Test/Spec/Reporter/XunitSpec.purs

```diff
@@ -17,24 +19,31 @@ xunitSpec = do
     describe "Spec" $
       describe "Reporter" $
         describe "Xunit" do
-          let doctype = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
           it "reports success" do
-            contents <- runXunit successSpec
+            let path1 = "output/success.tmp.xml"
+            contents <- runXunit successSpec path1
             contents `shouldContain` "<testcase name=\"works\"></testcase>"
           it "reports failure" do
-            contents <- runXunit failureSpec
+            let path2 = "output/error.tmp.xml"
+            contents <- runXunit failureSpec path2
             contents `shouldContain` "<testcase name=\"fails\">"
             contents `shouldContain` "Error"
 
   where
     successSpec = describe "a" (itOnly "works" (pure unit))
     failureSpec = describe "a" (itOnly "fails" (fail "OMG"))
 
-    runXunit spec = do
-      liftEffect $ do
-        let config = defaultConfig { exit = false }
-            path = "output/test.tmp.xml"
-        run' config [xunitReporter { indentation: 2, outputPath: path }] spec
-        contents <- readTextFile UTF8 path
-        unlink path
-        pure contents
+    -- Waits indefinitely for the file `fp` to appear, or until interrupted
+    waitForFile fp m = do
+      ex <- liftEffect $ exists fp
+      if ex
+        then pure unit
+        else delay m *> waitForFile fp m
```

Comment:
tried doing

```purescript
ifM (liftEffect $ exists fp) (pure unit) (delay m *> waitForFile fp m)
```

But that didn't work for some reason :/

Reply:
This does work for me :p